geps/gep-713/index.md

```diff
@@ -1,1157 +1,747 @@
+
+Meta resources follow a well-known structure that declares clearly a *target* and an *intent*.
+
+#### Meta resources are well-structured CRDs
+
+Meta resources are typically implemented as Custom Resource Definitions (CRDs) that comply with a particular structure. This structure includes fields for specifying references to one or more other objects－called “targets“－whose behavior the instances of the meta resource intend to augment (i.e. declare additional specification), along with resource-specific fields－the “spec proper”－to describe the intended augmented behavior.
+
+While the targets of a meta resource give the meta resource a *context*, the spec proper declares the *intent* within that context.
+
+A typical meta resource looks like the following – example provided based on a hypothetical `ColorPolicy` kind of meta resource:
+
+```yaml
+apiVersion: policies.controller.io/v1
+kind: ColorPolicy
+metadata:
+  name: my-color-policy
+spec:
+  targetRefs: ## target objects whose behaviour to augment
+  - group: gateway.networking.k8s.io/v1
+    kind: Gateway
+    name: my-gateway
+  rules:      ## the "spec proper" describing the intent, i.e. color the traffic blue
+    color: blue
+```
+
+#### Properties of meta resources
+
+##### Targetability (“where”)
+
+Meta resources specify one or more target resources or specific sections of resources, whose behavior the meta resource intends to augment.
+
+##### Semantics (“why”)
+
+Targeting a resource (or section of a resource) must be interpreted within a given semantics that is proper to the meta resource kind.
+
+Two different meta resource kinds that allow targeting resources of a same a given kind X may have very different semantics, not only because the purpose of the two meta resource kinds differ, but because the mechanics of calculating and applying the augment behavior for X can as well be very different between meta resource kinds.
+
+Often, the semantics of a meta resource is tightly coupled to the relationships and connections a target has with other kinds of objects, typically organized in a hierarchy of nested contexts. In this sense, targeting a given resource kind may have the semantics of spanning effect across yet other objects to which the target is related.
+
+##### Mergeability (“how”)
+
+Meta resources specify so-called *merge strategies* that dictate how multiple instances of the meta resource affecting a same resource (or section of a resource) should be dealt with in terms of calculating and enforcing a single set of rules specified by the meta resources.
 
-## Background and concepts
+The merge strategies typically include strategies for dealing with conflicting and/or missing specs, such as for applying default and/or override values on the target resources.
```

Comment:
I think it's important to note that sometimes the merge strategy may be specified in the design of the object (that is, it's a defaults policy or something), rather than in a field?

In fact, I tend to think that, if the merge strategy is listed in a field, it should be in the `status`, not the `spec`, since it's relevant info for _users_ of the Policy more than _implementers_ (who will build the merge strategy into code when handling the Policy anyway).

Reply:
+1, this feels like something that belongs in status.

Reply:
+1 that merge strategy may be defined in the metaresource, e.g. the API contract is only one is allowed per target, or multiple are allowed.

Reply:
I feel confused about the merge strategy in the `status` instead of the `spec`.

Are we talking about the metaresource's `status` and `spec`? Or the target's?

The merge strategy, if more than one is supported by the metaresource kind, is a choice of the user that declares an instance of the metaresource. How can it be in the status?

The user literally _specify_ what merge strategy to use when merging that instance of the metaresource. It should be in the `spec`, no?

Reply:
Added a few lines about merge strategy as a user choice or not, and reflected in the status stanza of the metaresource.